src/lib/systemd.ts

```diff
@@ -1,113 +1,130 @@
 import * as Bluebird from 'bluebird';
-import * as dbus from 'dbus-native';
+import * as dbus from 'dbus';
+import TypedError = require('typed-error');
 
-const bus = dbus.systemBus();
-const invokeAsync = Bluebird.promisify(bus.invoke).bind(bus);
+import log from './supervisor-console';
 
-const defaultPathInfo = {
-	path: '/org/freedesktop/systemd1',
-	destination: 'org.freedesktop.systemd1',
-	interface: 'org.freedesktop.systemd1.Manager',
-};
+export class DbusError extends TypedError {}
 
-const systemdManagerMethodCall = (
-	method: string,
-	signature?: string,
-	body?: any[],
-	pathInfo: {
-		path: string;
-		destination: string;
-		interface: string;
-	} = defaultPathInfo,
-) => {
-	if (signature == null) {
-		signature = '';
+const bus = dbus.getBus('system');
+const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
+	context: bus,
+});
+
+async function getSystemdInterface() {
+	try {
+		return await getInterfaceAsync(
+			'org.freedesktop.systemd1',
+			'/org/freedesktop/systemd1',
+			'org.freedesktop.systemd1.Manager',
+		);
+	} catch (e) {
+		throw new DbusError(e);
 	}
-	if (body == null) {
-		body = [];
+}
+
+export async function getLoginManagerInterface() {
+	try {
+		return await getInterfaceAsync(
+			'org.freedesktop.login1',
+			'/org/freedesktop/login1',
+			'org.freedesktop.login1.Manager',
+		);
+	} catch (e) {
+		throw new DbusError(e);
 	}
-	return invokeAsync({
-		...pathInfo,
-		member: method,
-		signature,
-		body,
-	});
-};
+}
 
```

Comment:
A helper like this could make the functions below shorter:

```typescript
async function callSystemdMethod(serviceName: string, ...args: any[]) {
    const systemd = await getSystemdInterface();
    try {
        systemd[serviceName](...args);
    } catch (e) {
        throw new DbusError(e);
    }
}
```

Also, maybe `const systemd = await getSystemdInterface();` could be called only once and stored somewhere.